### PR TITLE
[Solutions] Media Rendering Article

### PR DESCRIPTION
**Pull request by adamboeglin**
Added an article for media rendering using AWS Batch

### CODE REVIEW

File: docs/example-scenario/hpc/video-rendering.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+---
+title: 3D video rendering on Azure
+description: Running native HPC workloads in Azure using the Azure Batch service
+author: Adam Boeglin, Mike Warrington, and Ben Hummerstone
+ms.date: <publish or update date>
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: Populate the date please

File: docs/example-scenario/hpc/video-rendering.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+1. Upload input files and the applications to process those files to your Azure Storage account
+2. Create a Batch pool of compute nodes in your Batch account, a job to run the workload on the pool, and tasks in the job.
+3. Download input files and the applications to Batch
+4. Monitor task execution
+5. Upload task output
+6. Download output files
+
+To simplify this process, you could also use the [Batch Plugins for Maya & 3ds Max][batch-plugins]
+
+## Azure Batch: Creating an Azure Batch account and pools manually
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: make this an H3

File: docs/example-scenario/hpc/video-rendering.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+This sample solution is will provide help in learning how Azure Batch works while showcasing Azure Batch Labs as an example SaaS solution that can be developed for your own customers:
+
+[Azure Batch Masterclass][batch-labs-masterclass]
+
+## Azure Batch: Deploying a sample solution using an ARM (Azure Resource Manager) template
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: Make this an H3

File: docs/example-scenario/hpc/video-rendering.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+## Related Resources
+
+[Azure Batch Overview][batch-overview]
+
+[Azure Batch Documentation][batch-doc]
+
+[Using containers on Azure Batch][batch-containers]
+
+Other resources that are relevant that aren't linked from else where in the doc.
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: There is nothing below this, either remove it or move it to the top of the section

File: docs/example-scenario/hpc/video-rendering.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+
+### Scalability
+
+Pools within an Azure Batch account can either scale through manual intervention or, by using an formula based on Azure Batch metrics, be scaled automatically. For more information on this, please see the article
+[Create an automatic scaling formula for scaling nodes in a Batch pool][batch-scaling].
+
+### Security
+
+For a deeper discussion on [security][security] please see the relevant article in the architecure center.
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: For general guidance on designing secure solutions, see the [Azure Security Documentation][security].

File: docs/example-scenario/hpc/video-rendering.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+[Azure Batch Documentation][batch-doc]
+
+[Using containers on Azure Batch][batch-containers]
+
+Other resources that are relevant that aren't linked from else where in the doc.
+
+<!-- links -->
+[small-pricing]: https://azure.com/e/
+[medium-pricing]: https://azure.com/e/
+[large-pricing]: https://azure.com/e/
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: remove the above 3 lines

File: docs/example-scenario/hpc/video-rendering.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+[architecture]: ./media/native-hpc-ref-arch.png
+[resource-groups]: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-overview
+[security]: https://docs.microsoft.com/en-gb/azure/architecture/patterns/category/security
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: replace the link with the following: /azure/security/

File: docs/example-scenario/hpc/video-rendering.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+[batch-monitor]: https://docs.microsoft.com/en-us/azure/batch/monitoring-overview
+[batch-pricing]: https://azure.microsoft.com/en-gb/pricing/details/batch/
+[batch-doc]: https://docs.microsoft.com/en-us/azure/batch/
+[batch-overview]: https://azure.microsoft.com/en-us/services/batch/
+[batch-containers]: https://github.com/Azure/batch-shipyard
+[batch-rendering]: https://azure.microsoft.com/en-us/services/batch/rendering/
+[batch-saas]: https://customers.microsoft.com/en-us/story/insurance-tech-provider-delivers-disruptive-solutions
+[azure-arm-templates]: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-overview#template-deployment
+[batch-plugins]: https://docs.microsoft.com/en-us/azure/batch/batch-rendering-service#options-for-submitting-a-render-job
+#
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: remove

File: docs/example-scenario/hpc/video-rendering.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+[Azure Batch Overview][batch-overview]
+
+[Azure Batch Documentation][batch-doc]
+
+[Using containers on Azure Batch][batch-containers]
+
+Other resources that are relevant that aren't linked from else where in the doc.
+
+<!-- links -->
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: All of the links below that point to the docs site need to remove:  https://docs.microsoft.com/en-us; they should start /azure/

File: docs/example-scenario/hpc/video-rendering.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+author: Adam Boeglin, Mike Warrington, and Ben Hummerstone
+ms.date: <publish or update date>
+---
+# 3D video rendering on Azure
+
+3D rendering is a time consuming process that requires a significant amount of CPU time co complete.  On a single machine, the process of generating a video file from static assets can take hours or even days depending on the length and complexity of the video you are producing.  Many companies will purchase either expensive high end desktop computers to perform these tasks, or invest in large render farms that they can submit jobs to.  However, by taking advantage of Azure Batch, that power is available to you when you need it and shuts itself down when you don't, all without any capital investment.
+
+Batch gives you a consistent management experience and job scheduling, whether you select Windows Server or Linux compute nodes. With Batch, you can use your existing Windows or Linux applications, including AutoDesk Maya and Blender, to run large-scale render jobs in Azure.
+
+## Potential industry use cases
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: Related use cases

File: docs/example-scenario/hpc/video-rendering.md [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+
+Other potential use cases that would have a similar environment include:
+
+* 3D Modeling
+* Visual FX (VFX) Rendering
+* Video transcoding
+* Image processing, color correction, & resizing
+
+## Architecture diagram
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: Remove this H2, and move the diagram below into the Architecture section

File: docs/example-scenario/hpc/video-rendering.md [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+The solution diagram below is an example of this solution:
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: This can be deleted.

File: docs/example-scenario/hpc/video-rendering.md [hunk 13]
```diff
@@ -0,0 +1,187 @@
+---
+title: 3D video rendering on Azure
+description: Running native HPC workloads in Azure using the Azure Batch service
+author: Adam Boeglin, Mike Warrington, and Ben Hummerstone
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: This needs to be a GitHub username

File: docs/example-scenario/hpc/video-rendering.md [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+## Azure Batch: Deploying a sample solution using an ARM (Azure Resource Manager) template
+
+The following deployment will deploy:
+  - a new Azure Batch account
+  - a storage account
+  - a node pool associated with the batch account
+  - the node pool will be configured to use A2 v2 VMs with Canonical Ubuntu images
+  - the node pool will contain 0 VMs initially and will require scaling manually to add VMs
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmikewarr%2Farchitecture-center%2Fdocs%2Fsample-solutions%2Fhpc%2Fbatchcreatewithpools.json" target="_blank">
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: we need to move this template to the solutions-architecture repo, and update the link.

File: docs/example-scenario/hpc/video-rendering.md [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+    - NC, NCv2, NCv3, and ND sizes are optimized for compute-intensive and network-intensive applications and algorithms, including CUDA and OpenCL-based applications and simulations, AI, and Deep Learning. NV sizes are optimized and designed for remote visualization, streaming, gaming, encoding, and VDI scenarios utilizing frameworks such as OpenGL and DirectX.
+  - [Memory optmised VM sizes][compute-memory] When more memory is required, the memory optimized VM sizes offer a higher memory-to-CPU ratio.
+  - [General purposes VM sizes][compute-general] General purpose VM sizes are also available and provide balanced CPU-to-memory ratio.
+
+
+### Alternatives
+
+If you require more control over your rendering environment in Azure or need a hybrid implementation, then CycleCloud computing can help orchestrate an IaaS grid in the cloud. Using the same underlying Azure technologies as Azure Batch, it makes building and maintaining an IaaS grid an efficient process. To find out more and learn about the design principles, please use the following link:
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: There doesn't seem to be a following link, also it should read something like:

Review [design principles][design-principles] to learn how to...

and then add [design-principles]: at the bottom with the appropriate link.

File: docs/example-scenario/hpc/video-rendering.md [hunk 16]
```diff
@@ -0,0 +1,187 @@
...
+While there is currently no failover capability in Azure Batch, we recommend using the following steps to ensure availbility in case of an unplanned outage:
+
+* Create a Azure Batch account in an alternate Azure location with an alternate Storage Account
+* Create the same node pools with the same name, with 0 nodes allocated
+* Ensure Applications are created and updated to the alternate Storage Account
+* Upload input files and submit jobs to the alternate Azure Batch account
+
+## Pricing
+
+Explore the cost of running this solution, all of the services are pre-configured in the cost calculator.  To see how the pricing would change for your particular use case change the appropriate variables to match your expected needs.
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: Replace solution with scenario (where appropriate)

File: docs/example-scenario/hpc/video-rendering.md [hunk 17]
```diff
@@ -0,0 +1,187 @@
...
+  - [Memory optmised VM sizes][compute-memory] When more memory is required, the memory optimized VM sizes offer a higher memory-to-CPU ratio.
+  - [General purposes VM sizes][compute-general] General purpose VM sizes are also available and provide balanced CPU-to-memory ratio.
+
+
+### Alternatives
+
+If you require more control over your rendering environment in Azure or need a hybrid implementation, then CycleCloud computing can help orchestrate an IaaS grid in the cloud. Using the same underlying Azure technologies as Azure Batch, it makes building and maintaining an IaaS grid an efficient process. To find out more and learn about the design principles, please use the following link:
+
+For a complete overview of all the HPC solutions that are available to you in Azure, please see the article [HPC, Batch, and Big Compute solutions using Azure VMs][hpc-alt-solutions]
+
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: Add a Considerations H2

File: docs/example-scenario/hpc/video-rendering.md (hunk 2 above)
Comment (david-stanford) [reply docs/example-scenario/hpc/video-rendering.md]: actually both of these probably belong under a Deploy this scenario H2

**Comment by adamboeglin** [comment]
Code review items addressed, please review again.

File: docs/example-scenario/hpc/video-rendering.md [hunk 18]
```diff
@@ -0,0 +1,179 @@
...
+3. Download input files and the applications to Batch
+4. Monitor task execution
+5. Upload task output
+6. Download output files
+
+To simplify this process, you could also use the [Batch Plugins for Maya & 3ds Max][batch-plugins]
+
+## Deploy this scenario
+
+### Azure Batch: Creating an Azure Batch account and pools manually
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: Remove Azure Batch:

File: docs/example-scenario/hpc/video-rendering.md [hunk 19]
```diff
@@ -0,0 +1,179 @@
...
+This sample solution is will provide help in learning how Azure Batch works while showcasing Azure Batch Labs as an example SaaS solution that can be developed for your own customers:
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: This sample scenario will

File: docs/example-scenario/hpc/video-rendering.md [hunk 20]
```diff
@@ -0,0 +1,179 @@
...
+
+[Azure Batch Masterclass][batch-labs-masterclass]
+
+### Azure Batch: Deploying a sample solution using an ARM (Azure Resource Manager) template
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: Remove Azure Batch: 

Switch it to read: Deploying the sample scenario using an Azure Resource Manager (ARM) template

File: docs/example-scenario/hpc/video-rendering.md [hunk 21]
```diff
@@ -0,0 +1,179 @@
...
+
+The following deployment will deploy:
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: The template will deploy:

File: docs/example-scenario/hpc/video-rendering.md [hunk 22]
```diff
@@ -0,0 +1,179 @@
...
+[batch-scaling]: /azure/batch/batch-automatic-scaling
+[hpc-alt-solutions]: /azure/virtual-machines/linux/high-performance-computing?toc=%2fazure%2fbatch%2ftoc.json
+[batch-monitor]: /azure/batch/monitoring-overview
+[batch-pricing]: https://azure.microsoft.com/en-gb/pricing/details/batch/
+[batch-doc]: /azure/batch/
+[batch-overview]: https://azure.microsoft.com/services/batch/
+[batch-containers]: https://github.com/Azure/batch-shipyard
+[azure-arm-templates]: /azure/azure-resource-manager/resource-group-overview#template-deployment
+[batch-plugins]: /azure/batch/batch-rendering-service#options-for-submitting-a-render-job
+#
```
Comment (david-stanford) [inline docs/example-scenario/hpc/video-rendering.md]: still a random # sign

**Comment by adamboeglin** [comment]
Closing to merge into another branch.

**Pull request closed without merging**